adapters/genericListChecker/builder_test.go

```diff
@@ -26,119 +26,56 @@ func TestBuilderInvariants(t *testing.T) {
 	testutil.TestBuilderInvariants(b, t)
 }
 
-func TestWhitelistWithOverrides(t *testing.T) {
+func runTests(t *testing.T, bc BuilderConfig, ac AdapterConfig, goodValues []string, badValues []string) {
 	b := NewBuilder()
-	builderEntries := []string{"Four", "Five"}
-	b.Configure(&BuilderConfig{ListEntries: builderEntries, WhitelistMode: true})
+	b.Configure(&bc)
 
-	entries := []string{"One", "Two", "Three"}
-	config := AdapterConfig{ListEntries: entries}
-
-	aa, err := b.NewAdapter(&config)
+	aa, err := b.NewAdapter(&ac)
 	if err != nil {
 		t.Error("unable to create adapter " + err.Error())
 	}
 	a := aa.(adapters.ListChecker)
 
-	goodValues := []string{"One", "Two", "Three"}
 	for _, value := range goodValues {
-		var ok bool
-		ok, err = a.CheckList(value)
-		if !ok {
-			t.Error("Expecting check to pass")
+		ok, err := a.CheckList(value)
+		if !ok || err != nil {
+			t.Error("Expecting check to pass for value " + value)
```

Comment:
as elsewhere, i think the error message could be slightly improved to be:

`t.Errorf("CheckList(%s): got 'false'; want 'true', value)`